### ISSUE TITLE
Do not increment version if a devDep + peerDep?

### ISSUE THREAD

**Issue by milesj** [lerna/lerna#1898]
Say we have packages `foo`, `bar`, and `baz`, with `bar` and `baz` having `foo` as a peer dependency. Both of these packages also have a dev dependency for `foo` so that npm/yarn install/symlink everything correctly.

Now, when `foo` is updated and I run `lerna publish`, it increments the `foo` dev dependency in `bar` and `baz` (even when they haven't been modified), which causes a patch version to be published for these 2 packages. These patch versions are completely unnecessary.

It would be nice to avoid this if possible, possibly through a CLI flag.

## Expected Behavior
N/A, not sure it's scoped.

## Current Behavior
It unnecessarily releases versions when `devDeps` are updated.

## Possible Solution
Disable through a CLI flag? Don't increment `devDep` if also a `peerDep`?

## Context
This is causing unnecessary patch releases for packages.

## Your Environment

| Executable | Version |
| ---: | :--- |
| `lerna --version` | 3.10.7 |
| `npm --version`  | 6.7.0 |
| `yarn --version` | 1.13.0 |
| `node --version` | 10.14.1 |

**Comment by evocateur** [comment]
Yeah, this seems fair. I think we've long yearned for some sort of distinction between `dependencies` and `devDependencies` when determining the transitive bumps...

**Comment by diegohaz** [comment]
Is this related to #1737?

**Comment by evocateur** [comment]
@diegohaz Yes, this is essentially what I was agreeing with you in that thread.

Potential pitfalls, off the top of my head:
1. When a peer's paired devDep is bumped a major version, that _probably_ means the peer has been mutated, and we _should_ allow the transitive bump. (_which_ bump? ehhhh...) 
2. Reading minds is hard, so it probably needs to be an explicit prompt/option.

**Comment by alfonsomunozpomer** [comment]
This seems to be a duplicate of https://github.com/lerna/lerna/issues/2368.

**Comment by panayot-cankov** [comment]
Hello,

I have a similar issue on my end. I have a devDependency that generates some code and this code is reexported by several packages. I wish there was a mechanism to specify how the version change of one package affects the rest. E.g. somewhere in lerna confg to specify that MAJOR bump in foo would lead to MAJOR bump of baz.

Something like:
```
  "versionRelation": {
    "foo": {
      "baz": {
        "patch": "patch",
        "minor": "minor",
        "major": "major"
      }
    },
    "bar": {
      "baz": {
        "patch": "patch",
        "minor": "patch",
        "major": "patch"
      }
    }
  }
```

And that would basically mean that if lerna version figures out that "baz" was affected by "patch" it should bump the version of "foo" with a patch. And if the version of "baz" was affected by a "major" release. I think then bar and baz work as what is the default at the moment if baz was devDependency for bar.

And there may be few semantic one-liner options like:
```
  "foo": {
    "baz": "reexport" // meaning that code from baz behaves as reexported by foo, and breaking changes in baz lead inevitably to breaking changes in foo, but that's by repo's design and is ok
    "bar": "none" // bar may be a dev dependency in foo but is merely a linter, so no matter what we do to bar, it won't affect the public api and overall behavior in foo
  }
```

**Comment by leonheess** [comment]
Is this dead?
